test/wdio_functional/page_objects/management/index_pattern_page.js

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import BasePage from '../common/base_page';
+import { createTestSubjectSelectorCss } from '../../helpers/helpers';
+
+export default class IndexPatternPage extends BasePage {
+  constructor(driver) {
+    super(driver);
+    this.driver = driver;
+    this.createIndexPatternHeaderSelector = 'h1.euiTitle';
+    this.createIndexPatternNameInputSelector = createTestSubjectSelectorCss('createIndexPatternNameInput');
+    this.createIndexPatternStep2ButtonSelector = createTestSubjectSelectorCss('createIndexPatternGoToStep2Button');
+    this.createIndexPatternCreateButtonSelector = createTestSubjectSelectorCss('createIndexPatternCreateButton');
+    this.createIndexPatternTimeFieldSelectSelector = createTestSubjectSelectorCss('createIndexPatternTimeFieldSelect');
+    this.indexPatternsLinksSelector = createTestSubjectSelectorCss('indexPatternLink');
+    this.indexPatternTitleSelector = createTestSubjectSelectorCss('indexPatternTitle');
+    this.init();
+  }
+  init() {
+    expect('Create index pattern').toBe(this.getElementText(this.createIndexPatternHeaderSelector));
+    this.driver.waitUntil(() => {
+      return this.title === 'Kibana';
+    });
+    this.logger.debug('Step 2 button should not be enabled.');
+    expect(this.isEnabled(this.createIndexPatternStep2ButtonSelector)).toBe(false);
+  }
+
```

Comment:
It seems awkward to be to put this "test" in the init method.  If I think of a bunch more things I would want to check when navigating to the index pattern page I would expect to look for an "index pattern initial state" test, not an init method of the page object.

Reply:
The pattern is such that the object is created (or initialized) from navigating to the page. That's when the constructor is called. So the navigation and the creation of the object go hand in hand. This is why it makes sense for the `init()` function to house an assertion like this. I personally wouldn't have this assertion at all. I would just have a wait condition for that button to be enabled. But I included it because of our previous discussion on how this button has had issues in the past. 

See for reference: 
https://github.com/elastic/kibana/pull/20908/files#diff-fed29254eab9cb612b8634cc9356dde3R43